From 8273ed2ca1972d97feedf357109063da3b4cd5a5 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Mon, 1 Jan 2018 18:22:43 +0100
Subject: [PATCH 2/2] kltechnduo: Use rild2.libpath property for ril-daemon2

* And update rild.libpath to /vendor instead of /system/vendor

Change-Id: I362873e8eb65aee91c9cd617221ffab2f790a1a0
---
 init/init_klte.cpp       | 15 ++++++++++++---
 rootdir/etc/init.dsds.rc | 19 ++-----------------
 2 files changed, 14 insertions(+), 20 deletions(-)

diff --git a/init/init_klte.cpp b/init/init_klte.cpp
index 0439630..73f58a9 100644
--- a/init/init_klte.cpp
+++ b/init/init_klte.cpp
@@ -44,20 +44,29 @@ using android::init::property_set;
 
 void set_rild_libpath(char const *variant)
 {
-    std::string libpath("/system/vendor/lib/libsec-ril.");
+    std::string libpath("/vendor/lib/libsec-ril.");
     libpath += variant;
     libpath += ".so";
 
     property_override("rild.libpath", libpath.c_str());
 }
 
+void set_rild2_libpath(char const *variant)
+{
+    std::string libpath("/vendor/lib/libsec-ril-dsds.");
+    libpath += variant;
+    libpath += ".so";
+
+    property_set("rild2.libpath", libpath.c_str());
+}
+
 void gsm_properties(char const default_network[],
         char const *rild_lib_variant)
 {
     set_rild_libpath(rild_lib_variant);
+    set_rild2_libpath(rild_lib_variant);
 
     property_set("telephony.lteOnGsmDevice", "1");
-    property_set("rild.lib2_type", "gsm");
     property_set("ro.telephony.default_network", default_network);
 }
 
@@ -67,9 +76,9 @@ void cdma_properties(char const default_cdma_sub[], char const operator_numeric[
         char const *rild_lib_variant)
 {
     set_rild_libpath(rild_lib_variant);
+    set_rild2_libpath(rild_lib_variant);
 
     property_set("ril.subscription.types", "NV,RUIM");
-    property_set("rild.lib2_type", "cdma");
     property_set("ro.cdma.home.operator.numeric", operator_numeric);
     property_set("ro.cdma.home.operator.alpha", operator_alpha);
     property_set("ro.telephony.default_cdma_sub", default_cdma_sub);
diff --git a/rootdir/etc/init.dsds.rc b/rootdir/etc/init.dsds.rc
index a3303b2..3747d64 100644
--- a/rootdir/etc/init.dsds.rc
+++ b/rootdir/etc/init.dsds.rc
@@ -5,24 +5,9 @@
 # This is a common source of Android security bugs.
 #
 
-# Second RIL daemon for CDMA (SM-G9009W) device
-service ril-daemon2-cdma /vendor/bin/hw/rild -l /vendor/lib/libsec-ril-dsds.09w.so -c 2
+# Second RIL daemon for SIM2
+service ril-daemon2 /vendor/bin/hw/rild -l ${rild2.libpath} -c 2
     class main
     user radio
-    disabled
     group radio cache inet misc audio log readproc wakelock oem_2901
     capabilities BLOCK_SUSPEND NET_ADMIN NET_RAW
-
-# Second RIL daemon for GSM devices
-service ril-daemon2-gsm /vendor/bin/hw/rild -l /vendor/lib/libsec-ril-dsds.06w.so -c 2
-    class main
-    user radio
-    disabled
-    group radio cache inet misc audio log readproc wakelock oem_2901
-    capabilities BLOCK_SUSPEND NET_ADMIN NET_RAW
-
-on property:rild.lib2_type=cdma
-    start ril-daemon2-cdma
-
-on property:rild.lib2_type=gsm
-    start ril-daemon2-gsm
-- 
2.7.4

